extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/usecase/ContractAgreementPageService.java

```diff
@@ -0,0 +1,113 @@
+import de.sovity.edc.ext.wrapper.api.usecase.model.ContractAgreementPage;
+import de.sovity.edc.ext.wrapper.api.usecase.model.ContractNegotiationDto;
+import de.sovity.edc.ext.wrapper.api.usecase.model.TransferprocessDto;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.connector.contract.spi.negotiation.store.ContractNegotiationStore;
+import org.eclipse.edc.connector.contract.spi.types.agreement.ContractAgreement;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.spi.contractagreement.ContractAgreementService;
+import org.eclipse.edc.connector.spi.transferprocess.TransferProcessService;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.spi.asset.AssetIndex;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@RequiredArgsConstructor
+public class ContractAgreementPageService {
+    private final AssetIndex assetIndex;
+    private final ContractAgreementService contractAgreementService;
+    private final ContractNegotiationStore contractNegotiationStore;
+    private final TransferProcessService transferProcessService;
+
+    public ContractAgreementPage contractAgreementPage() {
+        var querySpec = QuerySpec.Builder.newInstance().build();
+        var contractAgreementDtos = new ArrayList<ContractAgreementDto>();
+        var contractAgreements = contractAgreementService.query(querySpec).getContent().toList();
+        var negotiations = getNegotiations().stream().collect(groupingBy(it -> it.getContractAgreement().getId()));
+
+        for (var contractAgreement : contractAgreements) {
+            var contractAgreementDtoDto = new ContractAgreementDtoDto(contractAgreement.getId());
+            var assetDto = getAssetDto(contractAgreement);
+            var policy = contractAgreement.getPolicy();
+            var transferProcessesDtos = getTransferprocessDtos(contractAgreement);
+            var contractNegotiationDtos = getContractNegotiationDtos(negotiations, contractAgreement);
+
+            contractAgreementDtos.add(new ContractAgreementDto(
+                    contractAgreementDtoDto, assetDto, policy, contractNegotiationDtos, transferProcessesDtos
+            ));
+        }
+
+        return new ContractAgreementPage(contractAgreementDtos);
+    }
+
+    private static ArrayList<ContractNegotiationDto> getContractNegotiationDtos(Map<String, List<ContractNegotiation>> negotiations, ContractAgreement contractAgreement) {
+        var agreementNegotiations = negotiations.getOrDefault(contractAgreement.getId(), List.of());
+        var contractNegotiationDtos = new ArrayList<ContractNegotiationDto>();
+
+        for (var agreementNegotiation : agreementNegotiations) {
```

Comment:
Schleifen wo möglich vermeiden, immer auf die Streams API setzen, bis man es mit Maschinennaher Algorithmik zu tun hat, Dynamischer Programmierung oder ähnlichem, wo man Arrays iteriert.

return negotiations.stream().map(it -> new ContractNegotiationDto(it.getId())).toList()